src/components/screens/vacation-requests-screen.tsx

```diff
@@ -282,15 +296,84 @@ const VacationRequestsScreen = () => {
     }
   };
 
+  /**
+   * Update a vacation request status
+   *
+   * @param newStatus vacation request status
+   * @param vacationRequestStatusId vacation request status id
+   */
+  const updateVacationRequestStatuses = async (
+    newStatus: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => {
+    if (!userProfile || !userProfile.id) return;
+
+    if (selectedRowIds && newStatus) {
+      const updatedStatuses: VacationRequestStatus[] = [];
+
+      await Promise.all(
+        selectedRowIds.map(async (selectedRowId) => {
+          const vacationRequestStatus = vacationRequestStatuses.find(
+            (vacationRequestStatus) => vacationRequestStatus.vacationRequestId === selectedRowId
+          );
+          try {
+            if (vacationRequestStatus?.id) {
+              const updatedStatus = await vacationRequestStatusApi.updateVacationRequestStatus({
+                id: vacationRequestStatus.id,
+                statusId: vacationRequestStatus.id,
+                vacationRequestStatus: {
+                  ...vacationRequestStatus,
+                  updatedAt: new Date(),
+                  status: newStatus
+                }
+              });
+              updatedStatuses.push(updatedStatus);
+            }
+          } catch (error) {
+            setError(`${strings.vacationRequestError.updateStatusError}, ${error}`);
+          }
+        })
+      );
+
+      const updatedVacationRequestStatuses = vacationRequestStatuses.map(
+        (vacationRequestStatus) => {
+          const foundUpdatedStatus = updatedStatuses.find(
+            (updatedStatus) => updatedStatus.id === vacationRequestStatus.id
+          );
+          if (foundUpdatedStatus && vacationRequestStatus.id === foundUpdatedStatus.id) {
+            return foundUpdatedStatus;
+          }
+          return vacationRequestStatus;
+        }
+      );
+      setVacationRequestStatuses(updatedVacationRequestStatuses);
+    }
+  };
+
   return (
-    <Card sx={{ margin: 0, padding: "10px", width: "100%", height: "100" }}>
-      <VacationRequestsTable
-        deleteVacationRequests={deleteVacationRequests}
-        createVacationRequest={createVacationRequest}
-        updateVacationRequest={updateVacationRequest}
-        loading={loading}
-      />
-    </Card>
+    <>
+      <Card sx={{ margin: 0, padding: "10px", width: "100%", height: "100", marginBottom: "16px" }}>
+        <VacationRequestsTable
+          deleteVacationRequests={deleteVacationRequests}
+          createVacationRequest={createVacationRequest}
+          updateVacationRequest={updateVacationRequest}
+          updateVacationRequestStatuses={updateVacationRequestStatuses}
+          loading={loading}
+        />
+      </Card>
+      <Card sx={{ margin: 0, padding: "10px", width: "100%" }}>
+        <Link
+          to={adminMode ? "/admin" : "/"}
+          state={{ keepData: true }}
```

Comment:
As mentioned in previous comments use atom or rename

Reply:
Renamed